File: client/src/app/modules/quotation/services/order.blockage.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { OrderBlockage } from '../../quotation/model/OrderBlockage';

@Injectable({
  providedIn: 'root'
})
export class OrderBlockageService extends AppRestService<OrderBlockage>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getOrderBlockages() {
    return this.getList(new HttpParams(), "order-blockages");
  }
  
   addOrUpdateOrderBlockage(orderBlockage: OrderBlockage) {
    return this.addOrUpdate(new HttpParams(), "order-blockage", orderBlockage, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/OrderBlockageService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.OrderBlockage;

public interface OrderBlockageService {
    public List<OrderBlockage> getOrderBlockages();

    public OrderBlockage getOrderBlockage(Integer id);

    public OrderBlockage addOrUpdateOrderBlockage(OrderBlockage orderBlockage);
}
